src/Calendar.js

```diff
@@ -18,390 +18,422 @@
-    hours = isPM ? (hours % 12) + 12 : (hours % 12);
-
-    if (!this._isTimeless()) {
-      this.date.hours(hours);
-      this.date.minutes(minutes);
+    if (params.$source) {
+      domClass.add(params.$source, 'selected');
+      if (domClass.contains(params.$source, 'isToday')) {
+        domClass.remove(this.todayButton, 'selected');
+      }
     }
 
-    this.updateDatetimeCaption();
-  },
-  toggleMeridiem: function toggleMeridiem(params) {
-    const el = params.$source;
-    const toggledValue = el && (domAttr.get(el, 'toggled') !== true);
-
-    if (el) {
-      domClass.toggle(el, 'toggleStateOn');
-      domAttr.set(el, 'toggled', toggledValue);
+    if (params.date) {
+      this.date.selectedDateMoment = moment(params.date, 'YYYY-MM-DD');
     }
 
-    this.updateDatetimeCaption();
-  },
-  populateSelector: function populateSelector(el, v, min, max) {
-    let val = v;
-    if (val > max) {
-      val = max;
+    if (this.date.selectedDateMoment.date() !== this.date.todayMoment.date()) {
+      domClass.add(this.todayButton, 'selected');
     }
-
-    el.options.length = 0;
-
-    for (let i = min; i <= max; i++) {
-      const opt = domConstruct.create('option', {
-        innerHTML: (this.monthNode === el) ? uCase(this.months[i]) : pad(i),
-        value: i,
-        selected: (i === val),
-      });
-
-      el.options[el.options.length] = opt;
+    if (this.date.month !== this.date.selectedDateMoment.month()) {
+      this.refreshCalendar(this.date);
     }
-  },
-  localizeViewTemplate: function localizeViewTemplate() {
-    const whichTemplate = arguments[0];
-    const formatIndex = arguments[1];
-    const fields = {
-      y: 'year',
-      Y: 'year',
-      M: 'month',
-      d: 'day',
-      D: 'day',
-      h: 'hour',
-      H: 'hour',
-      m: 'minute',
-    };
-
-    const whichField = fields[(formatIndex < 3) ? this.dateFormat.split(/[^a-z]/i)[formatIndex].charAt(0) : this.timeFormatText.replace(/[a]\s/i, '').split(/[^a-z]/i)[formatIndex - 3].charAt(0)];
-    const whichFormat = whichTemplate === 'selectorTemplate' ? whichField : uCase(whichField);
 
-    return string.substitute(this[whichTemplate], [whichFormat]);
+    return this;
   },
-  show: function show(o) {
-    this.inherited(arguments);
-    const options = o || this.options;
-
-    this.titleText = options.label ? options.label : this.titleText;
-
-    this.showTimePicker = this.options && this.options.showTimePicker;
-
-    this.date = moment((this.options && this.options.date) || moment());
-
-    if (this._isTimeless()) {
-      this.date.add({
-        minutes: this.date.zone(),
-      });
+  changeMonthShown: function changeMonthShown({ month }) {
+    domConstruct.empty(this.monthNode);
+    this.monthNode.innerHTML = month;
+    return this;
+  },
+  changeYearShown: function changeYearShown({ year }) {
+    domConstruct.empty(this.yearNode);
+    this.yearNode.innerHTML = ' ' + year;
```

Comment:
What is the purpose of the space here? Should we give the dom nodes margins via css instead of prefixing a space?